pkg/provisioner/ironic/ironic.go

```diff
@@ -1129,6 +1126,30 @@ func (p *ironicProvisioner) Adopt(force bool) (result provisioner.Result, err er
 	return
 }
 
+func (p *ironicProvisioner) ironicHasSameImage(ironicNode *nodes.Node) (sameImage bool) {
+	// To make it easier to test if ironic is configured with
+	// the same image we are trying to provision to the host.
+	if p.host.Spec.LiveImage != nil {
+		sameImage := (ironicNode.InstanceInfo["boot_iso"] == p.host.Spec.LiveImage.URL)
```

Comment:
This `:=` redefines `sameImage`, which I think shadows the named return value. I think you just want `=`.

Reply:
Oops - that's probably related to why it's not working!  

I wonder if we can modify the `got vet` config so we catch this kind of thing ref https://github.com/metal3-io/baremetal-operator/blob/master/.golangci.yml#L18